### DISCUSSION TITLE
License for Python Code editing

### DISCUSSION THREAD

**Discussion by sri-dor** [pybricks/support#1847] [Q&A] [answered]
It looks like Pybricks is free for Python. However, when I go to code.pybricks.com and try to create a Python file - I am not able to edit (it only allows read-only) and keeps popping a window for entering the license code for block coding. I am not looking for block coding - just Python. Can I do it at code.pybricks.com, or do I have to use VS Code/others for Python?

**Comment by BertLindeman** [comment, marked as answer]
Choose the python button for creating a new python source:
![image](https://github.com/user-attachments/assets/433fce0f-6130-4336-ac99-67db675e9ede)


Bert

**Reply by laurensvalk** [reply]
Thanks @BertLindeman  for answering.

@sri-dor -- A block program displays blocks and Python side by side. This Python code is not editable, since the blocks would override your changes and we wouldn't want people to lose their well intended changes. So I think you might be seeing this generated Python from a block program, and possibly you've dragged the divider between blocks and Python all the way left.

Indeed, selecting a Python-only program as @BertLindeman suggested is the solution.

**Comment by sri-dor** [comment]
Thank you! This worked.
One more question - when I import existing Python files, they are not editable. How do I mark them as Python files to allow editing?

**Reply by laurensvalk** [reply]
Existing Python files should be editable. Could you open one file with a text editor and paste the contents here so we could see what the issue is? 

Block programs are also just python files, but with one long line at the top that represents how each block is placed and configured.

If you have a previous block program but want to continue in Python, you can just delete that first line in your text editor, and then import it.

**Comment by sri-dor** [comment]
@laurensvalk Thank you so much for clarifying. It works as you described!
